### PR TITLE
Only get the configured principal's short name if kerberos is enabled

### PR DESCRIPTION
**Pull request by anwar6953**
Only get the configured principal's short name if kerberos is enabled.

The fix in https://github.com/caskdata/cdap/pull/6290 broke insecure clusters.

### CODE REVIEW

**Comment by anwar6953** [comment]
http://builds.cask.co/browse/CDAP-DUT4593-5

File: cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/DatasetServiceClient.java [hunk 1]
```diff
@@ -333,10 +328,11 @@ private HttpResponse doRequest(HttpMethod method, String resource,
       return builder;
     }
     String userId;
+    String currUserShortName = UserGroupInformation.getCurrentUser().getShortUserName();
     if (NamespaceId.SYSTEM.equals(namespaceId) &&
       // we compare short name, because in some YARN containers launched by CDAP, the current username isn't the full
       // configured principal
-      (!kerberosEnabled || UserGroupInformation.getCurrentUser().getShortUserName().equals(masterPrincipalShortName))) {
+      (!kerberosEnabled || currUserShortName.equals(new KerberosName(masterPrincipal).getShortName()))) {
```
Comment (chtyim) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/DatasetServiceClient.java]: You can do the `if` in the constructor, isn't it?

Comment (anwar6953) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/DatasetServiceClient.java]: And have a field like `isSystemCaller`?

Comment (chtyim) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/DatasetServiceClient.java]: More like `boolean masterUser`.

**Comment by chtyim** [comment]
LGTM

**Comment by anwar6953** [comment]
Squashed commits.

**Pull request merged**